lib/compat/wordpress-5.8/block-editor.php

```diff
@@ -69,26 +69,30 @@ function gutenberg_get_default_block_categories() {
  *
  * @since 10.5.0
  *
- * @param string|WP_Post $editor_name_or_post The name of the editor (e.g. 'post-editor')
- *                                            or the post object.
- *
+ * @param WP_Post|WP_Block_Editor_Context $post_or_block_editor_context The current post object or
+ *                                                                      the block editor context.
  * @return array[] Array of categories for block types.
  */
-function gutenberg_get_block_categories( $editor_name_or_post ) {
-	// Assume the post editor when the WP_Post object passed.
-	$editor_name        = is_object( $editor_name_or_post ) ? 'post-editor' : $editor_name_or_post;
-	$default_categories = gutenberg_get_default_block_categories();
+function gutenberg_get_block_categories( $post_or_block_editor_context ) {
```

Comment:
The argument is weird here, can't it be always the context?

Reply:
It's a preexisting function that accepted only a `$post` before. I share mutual feelings for the name 😄